### DISCUSSION TITLE
Guitar Effect Pedal

### DISCUSSION THREAD

**Discussion by Jerzeek** [pschatzmann/arduino-audio-tools#1434] [Ideas]
I was fascinated by the guitar effect example.
It has been 2 years since it was developed, it mostly stopped because of this bug: https://github.com/Ai-Thinker-Open/ESP32-A1S-AudioKit/issues/31, but recently someone shared a solution on Phills website: https://www.pschatzmann.ch/home/2021/12/15/the-ai-thinker-audiokit-audio-input-bug/
I tested this solution and for me, this worked perfectly!
Now my idea is to further build this guitar effect pedal. I can 3D model a nice case for the board, develop a system for a battery and speakers, and add some buttons or knows to also locally change the effects. 
I am looking for someone who can help me on the software side, I know too little about this. Currently, 3 effects are implemented but to my knowledge, there are many more in the library, maybe we can make a full-fledged example? and is there a way to control the effects with the onboard buttons?

**Comment by pschatzmann** [comment]
https://www.pschatzmann.ch/home/tag/guitarpedals/

**Comment by Jerzeek** [comment]
Yeah, this is great! and I got it up and running :)
Looking at your documentation it looks like there are more effects available, like ADSRGain, STKEcho, STKNReverb etc etc, all require some sort of input and would need to be added to the webUI, how can this be done?

**Reply by pschatzmann** [reply]
You should be able to find all necessary information in the class documentation. For the effects provided by the [STK library](https://github.com/pschatzmann/arduino-stk) you can also can also find further info there.

**Comment by Jerzeek** [comment]
![PXL_20240310_171415234 MP (Large)](https://github.com/pschatzmann/arduino-audio-tools/assets/22919620/4596ee17-fccc-4bbc-9ff0-499dce75ec5c)
![PXL_20240310_171532891 MP (Large)](https://github.com/pschatzmann/arduino-audio-tools/assets/22919620/7a9ee9b0-0f2d-4116-86a8-482da4ba1e64)
![PXL_20240310_171347857 MP (Large)](https://github.com/pschatzmann/arduino-audio-tools/assets/22919620/c0070c3b-5321-4ef2-830a-4074662f256e)
![PXL_20240310_171353630 MP (Large)](https://github.com/pschatzmann/arduino-audio-tools/assets/22919620/75721e8b-9321-4ea4-a7f7-412c8be24b83)
![PXL_20240310_171357795 MP (Large)](https://github.com/pschatzmann/arduino-audio-tools/assets/22919620/9200187e-382a-40cb-8e3f-f8ca8a25f416)

I made a little case for it, with an opening on the top for the headers.
[audiokit.zip](https://github.com/pschatzmann/arduino-audio-tools/files/14551374/audiokit.zip)
